packages/structure/test/unit/instanceAndUpdate.spec.js

```diff
@@ -93,19 +102,39 @@ describe('instantiating a structure', () => {
 
   describe('attributes initialization', () => {
     describe('default value', () => {
-      describe('when attribute default value is a static value', () => {
-        it('defaults to the static value', () => {
-          const user = new User();
+      describe('when attribute is non-nullable and has static default', () => {
+        describe('when passed value is undefined', () => {
+          it('defaults to the static value', () => {
+            const user = new User();
 
-          expect(user.name).toEqual('Name');
+            expect(user.name).toEqual('Name');
+          });
+        });
+
+        describe('when passed value is null', () => {
+          it('defaults to the static value', () => {
+            const user = new User({ name: null });
+
+            expect(user.name).toEqual('Name');
+          });
         });
       });
 
-      describe('when attribute default value is a function', () => {
-        it('calls the function using the instance of the object as parameter and perform coercion', () => {
-          const user = new User();
+      describe('when attribute is non-nullable and default value is a function', () => {
+        describe('when passed value is undefined', () => {
+          it('calls the function using the instance of the object as parameter and perform coercion', () => {
+            const user = new User();
+
+            expect(user.uuid).toEqual('10');
+          });
+        });
+
+        describe('when passed value is null', () => {
+          it('calls the function using the instance of the object as parameter and perform coercionl', () => {
```

Comment:
```suggestion
          it('calls the function using the instance of the object as parameter and perform coercion', () => {
```

I believe you made a small typo here :ok_woman:

Reply:
Oops, you're right!